libvast/src/transform_steps/delete.cpp

```diff
@@ -17,30 +17,58 @@
 
 namespace vast {
 
-delete_step::delete_step(const std::string& fieldname) : fieldname_(fieldname) {
+delete_step::delete_step(std::vector<std::string> fields)
+  : fields_(std::move(fields)) {
 }
 
-caf::expected<table_slice> delete_step::operator()(table_slice&& slice) const {
-  const auto& layout = slice.layout();
+caf::expected<std::pair<vast::type, std::vector<int>>>
+delete_step::adjust_layout(const vast::type& layout) const {
+  auto to_remove = std::set<vast::offset>{};
   const auto& layout_rt = caf::get<record_type>(layout);
-  auto offset = layout_rt.resolve_key(fieldname_);
-  if (!offset)
-    return std::move(slice);
-  auto columnn_index = layout_rt.flat_index(*offset);
-  auto adjusted_layout_rt
-    = layout_rt.transform({{*offset, record_type::drop()}});
+  for (const auto& key : fields_) {
+    auto offsets = layout_rt.resolve_key_suffix(key);
+    for (const auto& offset : offsets) {
+      to_remove.emplace(offset);
+    }
+  }
+  if (to_remove.empty())
+    return caf::no_error;
+  // adjust layout
```

Comment:
```suggestion
```

This comment doesn't add any meaning, so please omit it. This also applies to other similar comments in this diff.